### PR TITLE
[MDS-5193] - Implement Typescript

### PR DESCRIPTION
**Pull request by matbusby-fw**
### Had a little spare capacity so I took a stab at enabling Typescript

For now I only worked on the Minespace portion, but these concepts should work the same on Core.
Everything is working fine locally, and properly handling js, jsx, ts and tsx files
Updated the landing page to `LandingPage.tsx` and included some very light typescript
Updated `NoticeOfDeparture.js` to `NoticeOfDeparture.tsx`
updated jest to properly compile typescript pages
updated tests for these two components to .tsx

Wanted to throw a PR up at this point to get a couple of extra sets of eyes on this to see if anyone can see any issues or room for improvement.

### CODE REVIEW

File: services/minespace-web/package.json [hunk 1]
```diff
@@ -29,6 +29,8 @@
     "@syncfusion/ej2-react-filemanager": "19.1.54",
     "@syncfusion/ej2-react-pdfviewer": "19.4.55",
     "@syncfusion/ej2-splitbuttons": "19.1.54",
+    "@types/jest": "^29.1.0",
```
Comment (vyasworks) [inline services/minespace-web/package.json]: fix versions?

**Review by vyasworks — COMMENTED** [review]
Looks great! 

Would love to see this go to dev, are you able to make the changes on dockerfiles as well and see if it builds Mat?

File: services/minespace-web/package.json (hunk 1 above)
Comment (matbusby-fw) [reply services/minespace-web/package.json]: Ah, good catch.  Yes, I started trying to update all of the jest stuff to 29, but couldn't get it to work, so rolled the others back to 24.  Will update this.

**Comment by taraepp** [comment]
Exciting!

A couple of settings files that should be updated:
- .eslintrc.js should include ts/tsx files where we're configuring things for js files (the extensions array ~line 21)
- same with the root-level package.json for the lint-staged script ~line 34, 37 (lint-prepush is currently disabled, we could either remove or update that section too though)
- I think also the package.json files for core-web/minespace with the other scripts like format:check are just configured for js files
- some sonar-project.properties files (are we using this? I think so)
- I wonder about that exports.loadJS in services/core-web/webpack.parts.js (probably similar files in other directories)

**Comment by matbusby-fw** [comment]
Good input @taraepp.  I'll follow up on these this week at some point when I have the capacity!

File: services/minespace-web/package.json [hunk 2]
```diff
@@ -123,7 +132,6 @@
   },
   "optionalDependencies": {
     "enzyme-adapter-react-16": "1.14.0",
-    "jest": "24.8.0",
     "jest-canvas-mock": "2.1.0",
     "jest-environment-jsdom": "24.8.0",
     "jest-environment-jsdom-global": "1.2.0",
```
Comment (taraepp) [inline services/minespace-web/package.json:137]: I think the format scripts just below here (ln 150-152) and the equivalent bits in the core-web file are all that remains.

**Review by taraepp — APPROVED** [review]
🙌

**Pull request merged**